src/legacy/core_plugins/kibana/public/management/sections/objects/components/objects_table/objects_table.js

```diff
@@ -82,9 +82,7 @@ class ObjectsTableUI extends Component {
 
   constructor(props) {
     super(props);
-    this.savedObjectTypes = POSSIBLE_TYPES.filter(type => {
-      return this.props.uiCapabilities.savedObjectsManagement[type].read;
-    });
+    this.savedObjectTypes = this.props.uiCapabilities.savedObjectsManagement.read ? POSSIBLE_TYPES : [];
```

Comment:
We can simplify this further if we want: users shouldn't ever get to this screen (or use this component) if they don't have access to the Saved Objects Management feature. If they have access, then they're guaranteed to have the `read` privilege, currently.